pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -1000,66 +1000,31 @@ func (dr *DashboardServiceImpl) setDefaultPermissions(ctx context.Context, dto *
 		resource = "folder"
 	}
 
-	if !dr.cfg.RBAC.PermissionsOnCreation(resource) {
+	if dash.FolderUID != "" || !dr.cfg.RBAC.PermissionsOnCreation(resource) {
 		return
 	}
 
 	metrics.MFolderIDsServiceCount.WithLabelValues(metrics.Dashboard).Inc()
 
-	var permissions []accesscontrol.SetResourcePermissionCommand
-	if !provisioned && dto.User.IsIdentityType(claims.TypeUser, claims.TypeServiceAccount) {
-		userID, err := dto.User.GetInternalID()
-		if err != nil {
-			dr.log.Error("Could not make user admin", "dashboard", dash.Title, "id", dto.User.GetID(), "error", err)
-		} else {
-			permissions = append(permissions, accesscontrol.SetResourcePermissionCommand{
-				UserID: userID, Permission: dashboardaccess.PERMISSION_ADMIN.String(),
-			})
-		}
-	}
-
-	if dash.FolderUID == "" {
-		permissions = append(permissions, []accesscontrol.SetResourcePermissionCommand{
-			{BuiltinRole: string(org.RoleEditor), Permission: dashboardaccess.PERMISSION_EDIT.String()},
-			{BuiltinRole: string(org.RoleViewer), Permission: dashboardaccess.PERMISSION_VIEW.String()},
-		}...)
-	}
-
-	svc := dr.getPermissionsService(dash.IsFolder)
-	if _, err := svc.SetPermissions(ctx, dto.OrgID, dash.UID, permissions...); err != nil {
-		dr.log.Error("Could not set default permissions", "dashboard", dash.Title, "error", err)
+	permissions := []accesscontrol.SetResourcePermissionCommand{
+		{BuiltinRole: string(org.RoleEditor), Permission: dashboardaccess.PERMISSION_EDIT.String()},
+		{BuiltinRole: string(org.RoleViewer), Permission: dashboardaccess.PERMISSION_VIEW.String()},
 	}
-}
-
-func (dr *DashboardServiceImpl) setDefaultFolderPermissions(ctx context.Context, cmd *folder.CreateFolderCommand, f *folder.Folder, provisioned bool) {
```

Comment:
This change is not required, but I took the opportunity to remove `setDefaultFolderPermissions` as `setDefaultPermissions` method above can cover the folder use case as well.